process/process.go

```diff
@@ -302,13 +329,24 @@ func (p *Process) startRound(round block.Round) {
 			panic("fail to get previous block from storage")
 		}
 		messages := p.state.Precommits.QueryMessagesByHeightWithHighestRound(p.state.CurrentHeight - 1)
-		commits := make([]Precommit, 0, len(messages))
+		commits := make([]Precommit, 0, 2*p.state.Precommits.F()+1)
 		for _, message := range messages {
 			commit := message.(*Precommit)
 			if commit.blockHash.Equal(previousBlock.Hash()) {
 				commits = append(commits, *commit)
+				if len(commits) >= 2*p.state.Precommits.F()+1 {
+					// Restrict the len of commits to 2F+1, as is expected by
+					// the nodes that will be receiving this message.
+					break
+				}
+			}
+			if len(commits) == 2*p.state.Proposals.f+1 {
+				break
 			}
```

Comment:
In case there are more than 2F+1 precommits at that round. We only need to send 2F+1, so we only allow 2F+1 (this is in an effort to reduce network bandwidth).

Reply:
Isn't that already taken care of in the above check?

Reply:
That's true, missed that. Removed the check.